pymc3/distributions/distribution.py

```diff
@@ -56,17 +56,32 @@ def __new__(cls, name, *args, **kwargs):
                             "a 'with model:' block, or use the '.dist' syntax "
                             "for a standalone distribution.")
 
-        if isinstance(name, string_types):
-            data = kwargs.pop('observed', None)
-            cls.data = data
-            if isinstance(data, ObservedRV) or isinstance(data, FreeRV):
-                raise TypeError("observed needs to be data but got: {}".format(type(data)))
-            total_size = kwargs.pop('total_size', None)
-            dist = cls.dist(*args, **kwargs)
-            return model.Var(name, dist, data, total_size)
-        else:
+        if not isinstance(name, string_types):
             raise TypeError("Name needs to be a string but got: {}".format(name))
 
+        data = kwargs.pop('observed', None)
```

Comment:
Is it possible to add some comments explaining what the new code below does? Would be helpful for future-us and new comers I think.